src/kademlia/traversal_algorithm.cpp

```diff
@@ -521,11 +521,10 @@ void traversal_algorithm::add_router_entries()
 			, static_cast<void*>(this), int(std::distance(m_node.m_table.router_begin(), m_node.m_table.router_end())));
 	}
 #endif
-	for (auto i = m_node.m_table.router_begin()
-		, end(m_node.m_table.router_end()); i != end; ++i)
-	{
-		add_entry(node_id(), *i, observer::flag_initial);
-	}
+	std::for_each(m_node.m_table.router_begin()
```

Comment:
I don't think `routing_table` is iterable (it does not compile), maybe the names `router_begin` and `router_end` prevents that, maybe is a name change.